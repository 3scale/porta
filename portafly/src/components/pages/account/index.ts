export { default as AccountPage } from './AccountPage'
export * from './ApplicationsTabContent'
export * from './ApplicationsTabDataListTable'
export * from './OverviewTabContent'
export * from './ProductLink'
