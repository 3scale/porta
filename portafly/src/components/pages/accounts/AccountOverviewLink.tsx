import React from 'react'
import { IAccount } from 'types'
import { useTranslation } from 'i18n/useTranslation'
import { Button } from '@patternfly/react-core'

interface Props {
  account: IAccount
}

const AccountOverviewLink: React.FunctionComponent<Props> = ({ account }) => {
  const { t } = useTranslation('accountsIndex')

  return (
    <Button
      aria-label={t('accounts_table.account_overview_link_aria_label')}
      component="a"
      variant="link"
      href={`/accounts/${account.id}`}
      isInline
    >
      {account.orgName}
    </Button>
  )
}

export { AccountOverviewLink }
