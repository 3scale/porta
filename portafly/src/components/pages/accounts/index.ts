export { default as AccountsIndexPage } from './AccountsIndexPage'
export * from './AccountsTable'
export * from './AccountsDataListTable'
export * from './ActionButtonImpersonate'
export * from './CreateAccountButton'
export * from './ExportAccountsButton'
export * from './AccountOverviewLink'
export * from './utils'
