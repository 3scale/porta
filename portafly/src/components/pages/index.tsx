export * from './accounts'
export { default as Overview } from './Overview'
export * from './applications'
export { default as Login } from './Login'
export * from './plans'
export * from './product'
