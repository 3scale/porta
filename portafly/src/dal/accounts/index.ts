export * from './getDeveloperAccounts'
export * from './approveAccount'
