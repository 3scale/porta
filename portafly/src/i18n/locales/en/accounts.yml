---
page_title: Accounts | Portafly
page_title_description: Page title of the Accounts page
body_title: Accounts
body_title_description": Body title of the Accounts page
header_buttons:
  create_account: Create account
  create_account_description: Title of Create action button
  export_accounts: Export all accounts
  export_accounts_description: Title of Export all accounts button
accounts_table:
  empty_state_title: No results found
  empty_state_title_description: Title of the empty state of accounts table
  empty_state_body: No results match the filter criteria. Remove all filters or clear all filters to show results.
  empty_state_body_description: Body message of the empty state of accounts table
  empty_state_button: Clear all filters
  empty_state_button_description: Clear filters button of the empty state of accounts table
  data_toolbar:
    bulk_selector:
      none: Select none (0 items)
      none_description: Bulk Selector option for selecting no items
      page: Select page ({{count}} item)
      page_description: Bulk Selector option for selecting current page items
      page_plural: Select page ({{count}} items)
      page_plural_description: Bulk Selector option for selecting current page items (plural version)
      all: Select all ({{count}} item)
      all_description: Bulk Selector option for selecting all available items
      all_plural: Select all ({{count}} items)
      all_plural_description: Bulk Selector option for selecting all available items (plural version)
      label: "{{selectedCount}} selected"
      label_description: Label indicating how many items are currently selected
    bulk_actions:
      warning: Select accounts first to enable bulk actions
      warning_description: Warning that displays inside the actions dropdown, when no items are selected
      title: Actions
      title_description: Title of the Actions dropdown
      send_email: Send email
      send_email_description: Bulk action for Send main
      change_plan: Change account plan
      change_plan_description: Bulk action for Change plan
      change_state: Change state
      change_state_description: Bulk action for Change state
    search_widget:
      select_aria_label: Filter-by select
      select_aria_label_description: Aria label of the dropdown for filtering
      text_input_aria_label: Filter term input
      text_input_aria_label_description: Aria label of the text input for filtering
      select_state_aria_label: Select state input
      select_state_aria_label_description: Aria label of the select state input for filtering
      button_aria_label: Search button for input
      button_aria_label_description: Aria label of the button to filter
      placeholder: Filter by {{option}}
      placeholder_description: Placeholder for the text input
  aria_label: Accounts list
  aria_label_description: Aria label of the Accounts table component
  empty_state: There are no accounts
  empty_state_description: Message of the Accounts table empty state
  col_group: Group/organization
  col_group_description: Title of the Group/organization column
  col_admin: Admin
  col_admin_description: Title of the Admin column
  col_signup: Signup date
  col_signup_description: Title of the Signup date column
  col_plan: Plan
  col_plan_description: Title of the Plan column
  col_apps: Applications
  col_apps_description: Title of the Applications column
  col_state: State
  col_state_description: Title of the State column
  col_actions: Actions
  col_actions_description: Title of Actions column
  actions:
    impersonate: Act as
    approve: Approve
    activate: Activate
modals:
  send_email:
    title: Send email to admins of selected accounts
    to: To
    subject: Subject
    body: Body
    send: Send
    cancel: Cancel
  change_plan:
    title: Change plan for the selected accounts
    to: Selected accounts
    select_label: Select plan
    send: Change plan
    cancel: Cancel
  change_state:
    title: Change state for the selected accounts
    to: Selected accounts
    select_label: Select state
    select_helper_text: Changing the state of a developer account does not imply changing the state of its applications.
    send: Change state
    cancel: Cancel
  expand_list_button: "{{count}} more"
  expand_list_button_description: Button to expand the list ot items selected for this modal
toasts:
  send_email_start: Sending email...
  send_email_start_description: Alert message when email is being sent
  send_email_success: Email sent
  send_email_success_description: Alert message when email is sent
  send_email_error: Email could not be sent
  send_email_error_description: Alert message when email could not be sent
  change_plan_start: Changing accounts plan...
  change_plan_start_description: Alert message when plan is being changed
  change_plan_success: Account plan changed
  change_plan_success_description: Alert message when plan is changed
  change_plan_error: Account plan could not be changed
  change_plan_error_description: Alert message when plan could not be changed
  change_state_start: Changing accounts state...
  change_state_start_description: Alert message when state is being changed
  change_state_success: Account state changed
  change_state_success_description: Alert message when state is changed
  change_state_error: Account state could not be changed
  change_state_error_description: Alert message when state could not be changed
state:
  approved: Approved
  pending: Pending
  rejected: Rejected
  suspended: Suspended
