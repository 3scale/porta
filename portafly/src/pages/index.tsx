export { default as Overview } from './Overview'
export { default as Applications } from './Applications'
export { default as DeveloperAccounts } from './DeveloperAccounts'
export { default as Login } from './Login'
