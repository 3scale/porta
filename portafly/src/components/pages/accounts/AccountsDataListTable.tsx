import React from 'react'

import { IDeveloperAccount } from 'types'
import { DataListProvider, AccountsTable } from 'components'
import { generateColumns, generateRows } from 'components/pages/accounts/utils'
import { useTranslation } from 'i18n/useTranslation'

interface Props {
  accounts: IDeveloperAccount[]
}

const AccountsDataListTable: React.FunctionComponent<Props> = ({ accounts }) => {
  const { t } = useTranslation('accountsIndex')
  const columns = generateColumns(t)
  const rows = generateRows(accounts)

  const initialState = {
    data: accounts,
    table: { columns, rows }
  }

  return (
    <DataListProvider initialState={initialState}>
      <AccountsTable />
    </DataListProvider>
  )
}

export { AccountsDataListTable }
