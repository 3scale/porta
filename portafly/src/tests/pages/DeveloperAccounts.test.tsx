import React from 'react'

import { render } from 'tests/custom-render'
import { DeveloperAccounts } from 'pages'
import { useGetDeveloperAccounts } from 'dal/accounts'
import { developerAccounts } from 'tests/examples'

jest.mock('dal/accounts')

describe('when the request is pending', () => {
  (useGetDeveloperAccounts as jest.Mock).mockReturnValueOnce({ isPending: true })

  it('should render spinner', () => {
    const wrapper = render(<DeveloperAccounts />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('when backend returns an error', () => {
  (useGetDeveloperAccounts as jest.Mock).mockReturnValueOnce({ error: { message: 'ERROR' } })

  it('should render an alert', () => {
    const wrapper = render(<DeveloperAccounts />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('when backend returns a list of accounts', () => {
  (useGetDeveloperAccounts as jest.Mock).mockReturnValueOnce({ accounts: developerAccounts })

  it('should render a table with accounts', () => {
    const wrapper = render(<DeveloperAccounts />)
    expect(wrapper).toMatchSnapshot()
  })
})
