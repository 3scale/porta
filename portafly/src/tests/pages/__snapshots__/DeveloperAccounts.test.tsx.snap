// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when backend returns a list of accounts should render a table with accounts 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="pf-c-page__main-section pf-m-light"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            body_title
          </h1>
        </div>
      </section>
      <section
        class="pf-c-page__main-section"
      >
        <table
          aria-label="accounts_table.aria_label"
          class="pf-c-table pf-m-grid-md"
          data-ouia-component-id="1"
          data-ouia-component-type="PF4/Table"
          data-ouia-safe="true"
          role="grid"
        >
          <thead
            class=""
          >
            <tr>
              <th
                class=""
                data-key="0"
                data-label="accounts_table.col_group"
                scope="col"
              >
                accounts_table.col_group
              </th>
              <th
                class=""
                data-key="1"
                data-label="accounts_table.col_admin"
                scope="col"
              >
                accounts_table.col_admin
              </th>
              <th
                class=""
                data-key="2"
                data-label="accounts_table.col_signup"
                scope="col"
              >
                accounts_table.col_signup
              </th>
              <th
                class=""
                data-key="3"
                data-label="accounts_table.col_apps"
                scope="col"
              >
                accounts_table.col_apps
              </th>
              <th
                class=""
                data-key="4"
                data-label="accounts_table.col_state"
                scope="col"
              >
                accounts_table.col_state
              </th>
            </tr>
          </thead>
          <tbody
            class=""
          >
            <tr
              class=""
              data-ouia-component-id="2"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <td
                class=""
                data-key="0"
                data-label="accounts_table.col_group"
              >
                Josemi's Corp.
              </td>
              <td
                class=""
                data-key="1"
                data-label="accounts_table.col_admin"
              >
                Josemi
              </td>
              <td
                class=""
                data-key="2"
                data-label="accounts_table.col_signup"
              >
                2019-10-18T05:13:26Z
              </td>
              <td
                class=""
                data-key="3"
                data-label="accounts_table.col_apps"
              >
                3
              </td>
              <td
                class=""
                data-key="4"
                data-label="accounts_table.col_state"
              >
                approved
              </td>
            </tr>
            <tr
              class=""
              data-ouia-component-id="3"
              data-ouia-component-type="PF4/TableRow"
              data-ouia-safe="true"
            >
              <td
                class=""
                data-key="0"
                data-label="accounts_table.col_group"
              >
                Damian's Corp.
              </td>
              <td
                class=""
                data-key="1"
                data-label="accounts_table.col_admin"
              >
                Damian
              </td>
              <td
                class=""
                data-key="2"
                data-label="accounts_table.col_signup"
              >
                2019-10-18T05:13:26Z
              </td>
              <td
                class=""
                data-key="3"
                data-label="accounts_table.col_apps"
              >
                3
              </td>
              <td
                class=""
                data-key="4"
                data-label="accounts_table.col_state"
              >
                approved
              </td>
            </tr>
          </tbody>
        </table>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="pf-c-page__main-section pf-m-light"
    >
      <div
        class="pf-c-content"
      >
        <h1
          class=""
          data-pf-content="true"
        >
          body_title
        </h1>
      </div>
    </section>
    <section
      class="pf-c-page__main-section"
    >
      <table
        aria-label="accounts_table.aria_label"
        class="pf-c-table pf-m-grid-md"
        data-ouia-component-id="1"
        data-ouia-component-type="PF4/Table"
        data-ouia-safe="true"
        role="grid"
      >
        <thead
          class=""
        >
          <tr>
            <th
              class=""
              data-key="0"
              data-label="accounts_table.col_group"
              scope="col"
            >
              accounts_table.col_group
            </th>
            <th
              class=""
              data-key="1"
              data-label="accounts_table.col_admin"
              scope="col"
            >
              accounts_table.col_admin
            </th>
            <th
              class=""
              data-key="2"
              data-label="accounts_table.col_signup"
              scope="col"
            >
              accounts_table.col_signup
            </th>
            <th
              class=""
              data-key="3"
              data-label="accounts_table.col_apps"
              scope="col"
            >
              accounts_table.col_apps
            </th>
            <th
              class=""
              data-key="4"
              data-label="accounts_table.col_state"
              scope="col"
            >
              accounts_table.col_state
            </th>
          </tr>
        </thead>
        <tbody
          class=""
        >
          <tr
            class=""
            data-ouia-component-id="2"
            data-ouia-component-type="PF4/TableRow"
            data-ouia-safe="true"
          >
            <td
              class=""
              data-key="0"
              data-label="accounts_table.col_group"
            >
              Josemi's Corp.
            </td>
            <td
              class=""
              data-key="1"
              data-label="accounts_table.col_admin"
            >
              Josemi
            </td>
            <td
              class=""
              data-key="2"
              data-label="accounts_table.col_signup"
            >
              2019-10-18T05:13:26Z
            </td>
            <td
              class=""
              data-key="3"
              data-label="accounts_table.col_apps"
            >
              3
            </td>
            <td
              class=""
              data-key="4"
              data-label="accounts_table.col_state"
            >
              approved
            </td>
          </tr>
          <tr
            class=""
            data-ouia-component-id="3"
            data-ouia-component-type="PF4/TableRow"
            data-ouia-safe="true"
          >
            <td
              class=""
              data-key="0"
              data-label="accounts_table.col_group"
            >
              Damian's Corp.
            </td>
            <td
              class=""
              data-key="1"
              data-label="accounts_table.col_admin"
            >
              Damian
            </td>
            <td
              class=""
              data-key="2"
              data-label="accounts_table.col_signup"
            >
              2019-10-18T05:13:26Z
            </td>
            <td
              class=""
              data-key="3"
              data-label="accounts_table.col_apps"
            >
              3
            </td>
            <td
              class=""
              data-key="4"
              data-label="accounts_table.col_state"
            >
              approved
            </td>
          </tr>
        </tbody>
      </table>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`when backend returns an error should render an alert 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="pf-c-page__main-section pf-m-light"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            body_title
          </h1>
        </div>
      </section>
      <section
        class="pf-c-page__main-section"
      >
        <div
          aria-label="Danger Alert"
          class="pf-c-alert pf-m-danger"
          data-ouia-component-id="0"
          data-ouia-component-type="PF4/Alert"
          data-ouia-safe="true"
        >
          <div
            class="pf-c-alert__icon"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                transform=""
              />
            </svg>
          </div>
          <h4
            class="pf-c-alert__title"
          >
            <span
              class="pf-u-screen-reader"
            >
              Danger alert:
            </span>
            ERROR
          </h4>
          
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="pf-c-page__main-section pf-m-light"
    >
      <div
        class="pf-c-content"
      >
        <h1
          class=""
          data-pf-content="true"
        >
          body_title
        </h1>
      </div>
    </section>
    <section
      class="pf-c-page__main-section"
    >
      <div
        aria-label="Danger Alert"
        class="pf-c-alert pf-m-danger"
        data-ouia-component-id="0"
        data-ouia-component-type="PF4/Alert"
        data-ouia-safe="true"
      >
        <div
          class="pf-c-alert__icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              transform=""
            />
          </svg>
        </div>
        <h4
          class="pf-c-alert__title"
        >
          <span
            class="pf-u-screen-reader"
          >
            Danger alert:
          </span>
          ERROR
        </h4>
        
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`when the request is pending should render spinner 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="pf-c-page__main-section pf-m-light"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            body_title
          </h1>
        </div>
      </section>
      <section
        class="pf-c-page__main-section"
      >
        <section
          aria-label="Loading Content Container"
          class="pf-c-page__main-section"
        >
          <div
            class="pf-c-empty-state"
          >
            <div
              class="pf-c-empty-state__content"
            >
              <div
                class="pf-c-empty-state__icon"
              >
                <span
                  aria-valuetext="Loading..."
                  class="pf-c-spinner pf-m-xl"
                  role="progressbar"
                >
                  <span
                    class="pf-c-spinner__clipper"
                  />
                  <span
                    class="pf-c-spinner__lead-ball"
                  />
                  <span
                    class="pf-c-spinner__tail-ball"
                  />
                </span>
              </div>
              <h2
                class="pf-c-title pf-m-lg"
              >
                Loading
              </h2>
            </div>
          </div>
        </section>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="pf-c-page__main-section pf-m-light"
    >
      <div
        class="pf-c-content"
      >
        <h1
          class=""
          data-pf-content="true"
        >
          body_title
        </h1>
      </div>
    </section>
    <section
      class="pf-c-page__main-section"
    >
      <section
        aria-label="Loading Content Container"
        class="pf-c-page__main-section"
      >
        <div
          class="pf-c-empty-state"
        >
          <div
            class="pf-c-empty-state__content"
          >
            <div
              class="pf-c-empty-state__icon"
            >
              <span
                aria-valuetext="Loading..."
                class="pf-c-spinner pf-m-xl"
                role="progressbar"
              >
                <span
                  class="pf-c-spinner__clipper"
                />
                <span
                  class="pf-c-spinner__lead-ball"
                />
                <span
                  class="pf-c-spinner__tail-ball"
                />
              </span>
            </div>
            <h2
              class="pf-c-title pf-m-lg"
            >
              Loading
            </h2>
          </div>
        </div>
      </section>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
